common/src/main/java/feast/common/logging/AuditLogger.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.common.logging;
+
+import com.google.protobuf.Message;
+import feast.common.logging.config.LoggingProperties;
+import feast.common.logging.config.LoggingProperties.AuditLogProperties;
+import feast.common.logging.entry.ActionAuditLogEntry;
+import feast.common.logging.entry.AuditLogEntry;
+import feast.common.logging.entry.AuditLogEntry.FeastInstance;
+import feast.common.logging.entry.AuditLogEntryKind;
+import feast.common.logging.entry.LogResource;
+import feast.common.logging.entry.LogResource.ResourceType;
+import feast.common.logging.entry.MessageAuditLogEntry;
+import feast.common.logging.entry.TransitionAuditLogEntry;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+import org.slf4j.event.Level;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
```

Comment:
I'm not expert in SLF4J, but wouldnt creating the logger here cause it to always use this class AuditLogger as the place of logging. Isn't it better to have this AuditLogger return objects from its methods and to have the logger defined in the calling class?

Reply:
Also, by wrapping the logging class we remove control from developers. Every time they want to add custom functionality they have to come and hack this class.

Reply:
> I'm not expert in SLF4J, but wouldnt creating the logger here cause it to always use this class AuditLogger as the logging class (as opposed to the class that calls the logger)? 
>  Also, by wrapping the logging class we remove control from developers. Every time they want to add custom functionality they have to come and hack this class.

One way we can workaround is to allow the user to provide their own logger when logging.

>Isn't it better to have this AuditLogger return objects from its methods and to have the logger defined in the calling class?

The `AuditLogger` serves the following purposes:
- tags the logs with `AUDIT_MARK` marker, which is used by [log4j2 logging backend]  to determine the right logging pattern to render the audit logs in.
- ability to transparently config audit logger.  (ie enable/disable message level audit logging.) from `application.yml`

Reply:
Makes sense. We can keep it for now.